client/state/sites/blog-stickers/reducer.js

```diff
@@ -0,0 +1,25 @@
+/**
+ * External dependencies
+ */
+import { merge } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { SITES_BLOG_STICKER_LIST_RECEIVE } from 'state/action-types';
+import { combineReducers, createReducer } from 'state/utils';
+
+export const items = createReducer(
+	{},
+	{
+		[ SITES_BLOG_STICKER_LIST_RECEIVE ]: ( state, action ) => {
+			return merge( {}, state, {
+				[ action.payload.blogId ]: action.payload.stickers,
```

Comment:
You would need to modify the action creator to have `blogId` as a top level right beside payload instead of inside it